### PR TITLE
[FIX] spreadsheet: filter creation/deletion reloads datasources

### PR DESCRIPTION
**Pull request by rrahir**
The domain of the datasources was not recomputed on the deletion of a filter that impacted it. Furthermore, the issue was also present when a client would UNDO/REDO some revisions.

Task 3002004

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js [hunk 1]
```diff
@@ -69,9 +69,6 @@ export default class PivotCorePlugin extends CorePlugin {
                     return CommandResult.InvalidNextId;
                 }
                 break;
-            case "REMOVE_GLOBAL_FILTER":
```
Comment (LucasLefevre) [inline addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js:72]: why is this removed?

File: addons/spreadsheet/static/tests/global_filters/global_filters_chart_test.js [hunk 2]
```diff
@@ -107,6 +107,11 @@ QUnit.module("spreadsheet > Global filters chart", {}, () => {
             type: "date",
         };
         assert.deepEqual(model.getters.getChartFieldMatch(chartId)[filter.id], matching);
+        assert.deepEqual(model.getters.getChartDataSource(chartId)._customDomain, [
```
Comment (LucasLefevre) [inline addons/spreadsheet/static/tests/global_filters/global_filters_chart_test.js:110]:
```suggestion
        assert.deepEqual(model.getters.getChartDataSource(chartId).getComputedDomain(), [
```
You can use the public API instead of the private attribute :)

File: addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js (hunk 1 above)
Comment (rrahir) [reply addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js:72]: because it's in the allowDispatch :D  it was a duplication of the function `_setPivotFieldMatching` that I introduced in the refactoring and I managed to find the existence of the anomally through one of the tests but can't really tell you which one :/  Most likely, the fix worked for every ds except pivots because their history was corrupted by this line

**Comment by rrahir** [comment]
@LucasLefevre Updated :)

**Review by LucasLefevre — APPROVED** [review]
robodoo r+
